### PR TITLE
Complete home page

### PR DESCRIPTION
**Pull request by SaurabhChawda**
Added Complete Home-page with functionality.
It Includes:-
- Seach bar:- where users can search quizzes by category.
- card:- card contain quiz category.
Screenshot (323)

### CODE REVIEW

File: src/Context/DataContext.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const DataContext = createContext(null);
+
+const DataProvider = ({ children }) => {
+  const [data, setdata] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        await axios
+          .get("/api/data")
+          .then((response) => setdata(response.data.data));
```
Comment (JantuDeb) [inline src/Context/DataContext.jsx]:
```
const {data} = await axios.get("api/data")
```

**Review by JantuDeb — APPROVED** [review]
It's looks good, I have pointed out an issue do check it out

**Review by ShreyashMahajan — APPROVED** [review]
looks good to merge

File: src/Context/DataContext.jsx (hunk 1 above)
Comment (SaurabhChawda) [reply src/Context/DataContext.jsx]: Updated, Thankyou

**Comment by SaurabhChawda** [comment]
> looks good to merge

ThankYou

**Pull request merged**